### PR TITLE
[UI] Refactor Banks Page

### PR DESCRIPTION
**Pull request by deannaflare**
Summary
---------

This PR refactors the bank grid page, match form, create bank form, and add content form

- the banks are now listed in a table instead of a grid of cards
- The Match Form is now in a card with different styling
- the Add new bank form is now in a modal
- the add content form is now in a modal that is triggered by a button per item in the table (there is also now a success state for it)

Test Plan

- create a new bank and verify that it adds like expected
- add content to banks and verify that you see a success state as expected with a hash value returned
- Search by that same content and verify that you see the matches you expect (matches will highlight in the table)

---------

Screenshots:

Screenshot 2024-10-04 at 9 19 32 AM
Screenshot 2024-10-04 at 9 09 38 AM
Screenshot 2024-10-04 at 9 10 05 AM
Screenshot 2024-10-04 at 9 10 19 AM
Screenshot 2024-10-04 at 9 10 33 AM

### CODE REVIEW

**Comment by deannaflare** [comment]
Buggy, ugly code but I'm going to the pub

here's a start

File: hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/banks_grid.html.j2 [hunk 1]
```diff
@@ -0,0 +1,121 @@
+{# Copyright (c) Meta Platforms, Inc. and affiliates. #}
+<h1>Banks</h1>
+<p class="text-secondary">Banks are like folders. You can do the following on this page:</p>
+<ul class="text-secondary">
+    <li>Create new banks</li>
+    <li>Add content: Add content to each bank</li>
+    <li>Search by content: Find banks that contain a certain piece of content</li>
+</ul>
```
Comment (deannaflare) [inline hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/banks_grid.html.j2:8]: I made this up lol. Someone should come up with better copy to describe these concepts.

I think all the pages should have a h1 title and a description

File: hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/match_form.html.j2 [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                </div>
+        `
+        // Append both the banks and hashes sections to the "matches" element
+        matches.innerHTML = content;
+    }
+
+    const highlightMatchedResults = (result) => {
+        // Render matched banks
+        result.banks.forEach((bankName) => {
+            document.getElementById(`bank_item_${bankName}`).classList.add("bg-light");
```
Comment (deannaflare) [inline hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/match_form.html.j2:97]: I wish this was a darker highlight, but I don't want to mess around with adding custom colors to this lol. I'll leave it up to ya'll

File: hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/banks_grid.html.j2 [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            <tr id="bank_item_{{bank['name']}}">
+                <th scope="row" style="background-color: transparent;">{{ loop.index }}</th>
+                <td style="background-color: transparent;">{{ bank['name'] }}</td>
+                <td style="background-color: transparent;">{{ bank['matching_enabled_ratio'] }}</td>
+                <td style="background-color: transparent;">
+                    {% with bank_title=bank['name'] %}
+                        {% include 'components/banks/add_to_bank_form.html.j2' %}
+                    {% endwith %}
+                </td>
+            </tr>
```
Comment (deannaflare) [inline hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/banks_grid.html.j2:32]: eventually, these will link to their own page with stats and such.

Leaving that up to ya'll too

File: hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/add_to_bank_form.html.j2 [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{# Copyright (c) Meta Platforms, Inc. and affiliates. #}
+<button type="button" class="btn btn-light" data-bs-toggle="modal" data-bs-target="#{{bank_title}}">Add content</button>
```
Comment (deannaflare) [inline hasher-matcher-actioner/src/OpenMediaMatch/templates/components/banks/add_to_bank_form.html.j2:2]: I wish this was an icon and not "Add content" but I didn't want to mess around with icon libraries either lol

**Review by Dcallies — APPROVED** [review]
We usually break something with these UI restyles, but it's labor intensive to click on everything, and it looks like you did all the tests I would have run. Let's land-and-iterate.

**Pull request merged**